prov/mrail/src/mrail_cq.c

```diff
@@ -393,12 +393,14 @@ void mrail_poll_cq(struct util_cq *cq)
 			util_buf_release(tx_buf->ep->tx_buf_pool, tx_buf);
 			ofi_ep_lock_release(&tx_buf->ep->util_ep);
 
-			ret = ofi_cq_write(cq, comp.op_context, comp.flags,
-					   0, NULL, 0, 0);
-			if (ret) {
-				FI_WARN(&mrail_prov, FI_LOG_CQ,
-					"Unable to write to util cq\n");
-				goto err;
+			if (tx_buf->flags & FI_COMPLETION) {
```

Comment:
tx_buf has been released above. Is it safe to re-use it here?

Reply:
Good catch! I'll move it down.